Adds GeGLU activation and tests.

HMUNACHI opened the pull request:
# What does this PR do?

<!--

This PR adds the GeGLU (Gated Linear Unit with GELU) activation function to the activation module of the Flax library. It is an increasingly popular activation layer which combines a linear transformation with a GELU activation in a gating mechanism, balancing linearity and non-linearity.

GeGLU is parameterised and as such uses Flax's Dense layer and was not part of Jax itself. The implementation allows for an optional output_dim attribute, which can be used to specify the number of output features. This can increase or decrease the -1th dimensions of the results and the code tries to account for that. The tests validate the functionality in three scenarios: standard usage, output dimension expansion, and output dimension contraction, ensuring the layer's reliability in various use cases.
-->

Fixes # (3483)

## Checklist
- [x] This PR adds the GeGLU (Gated Linear Unit with GELU) activation function to the activation module.
- [x] This change is discussed in a Github issue/
      [discussion](https://github.com/google/flax/discussions/3483)
- [x] The documentation and docstrings adhere to the
      [documentation guidelines](https://github.com/google/flax/blob/main/docs/README.md#how-to-write-code-documentation).
- [x] This change includes necessary high-coverage tests.
      (No quality testing = no merge!)

chiamp commented:
Thanks for adding this! Could you also add a test to check the GeGLU output value with an explicit, manual calculation of what the expected output value should be?

HMUNACHI commented:
I have added more tests as requested. I noticed PReLU did not have those, so I added such tests for it too.

HMUNACHI commented:
> Thanks for adding this! Could you also add a test to check the GeGLU output value with an explicit, manual calculation of what the expected output value should be?

I have added more tests as requested. I noticed PReLU did not have those, so I added such tests for it too. Thanks.

HMUNACHI commented:
> Thanks for adding the tests! We actually have a separate, incoming [PR](https://github.com/google/flax/pull/3498) for the `PReLU` test, so could you remove that test?

Thanks, I have removed the PRelu tests as requested.

chiamp commented:
Thanks! Looks like there's a merge conflict and CI is failing. Could you also rebase and squash your commits?
